src/containers/course-section/resources-list/ResourcesList.tsx

```diff
@@ -56,6 +56,21 @@ const ResourcesList: FC<ResourcesListProps> = ({
     setResources(reorderedItems)
   }
 
+  const setResourceAvailability = (
+    id: string,
+    availability: ResourceAvailabilityStatusEnum,
+    openFromDate?: string | null
+  ) => {
+    const resources = [...items]
+    const resource = resources.find((item) => item._id === id)
+
+    if (resource) {
+      resource.resourceAvailability = availability
+      openFromDate !== undefined && (resource.openFromDate = openFromDate)
```

Comment:
do you need to check `openFromDate !== undefined` ?
I think, `openFromDate && (resource.openFromDate = openFromDate)` should work

Reply:
Actually I do. `openFromDate` can be null and we need to distinguish between null and undefined. The solution you are suggesting will not set the null value for the date.

Reply:
ok, but I don`t see the place where it can set undefined to openFromDate, as I see according to code and interface in ResourceItem where you passed this function, openFromDate can be null or string but not undefined, or I am wrong?

Reply:
I did that in case we will need an opportunity to call the function without specifying date at all. But yeah, I don't think I use it in that way so I will remove the check.

Reply:
Done